lib/msal-browser/src/app/ClientApplication.ts

```diff
@@ -620,15 +622,19 @@ export abstract class ClientApplication {
      */
     protected async acquireTokenByRefreshToken(
         commonRequest: CommonSilentFlowRequest,
-        silentRequest: SilentRequest
+        silentRequest: SilentRequest,
+        preQueueTime?: number
     ): Promise<AuthenticationResult> {
+        this.performanceClient.addQueueMeasurement(PerformanceEvents.AcquireTokenByRefreshToken, commonRequest.correlationId, preQueueTime);
         switch(silentRequest.cacheLookupPolicy) {
             case CacheLookupPolicy.Default:
             case CacheLookupPolicy.AccessTokenAndRefreshToken:
             case CacheLookupPolicy.RefreshToken:
             case CacheLookupPolicy.RefreshTokenAndNetwork:
                 const silentRefreshClient = this.createSilentRefreshClient(commonRequest.correlationId);
-                return silentRefreshClient.acquireToken(commonRequest);
+
+                const preAcquireTokenTime = this.performanceClient.getCurrentTime();
+                return silentRefreshClient.acquireToken(commonRequest, preAcquireTokenTime);
```

Comment:
The general pattern you have here is
`currTime = getCurrentTime();`
`method(request, currTime);`


If you added a timestamp or map of timestamps to the request object could we eliminate all the prototype changes to not add the currTime parameter?  Code would look a bit more like this:
`request.currentTime = getCurrentTime();`
`method(request);`

Reply:
Updated timestamp method to store current time on the Performance client. Please see updated pattern.